### PR TITLE
Feature/items

### PR DESCRIPTION
**Pull request by macbaszii**
[WIP] [Not Urgent]

@kittinunf Could you help me why I always get error code 3840 when fetch all items ? I'm going to bed now. get stuck for 15 minutes.

### CODE REVIEW

**Comment by kittinunf** [comment]
Fixed! by   b2e6629

File: SecretListServer/SecretListServer/Auth.swift [hunk 1]
```diff
@@ -12,17 +12,49 @@ import Swifter
...
+        return .OK(.Json(json))
+    }
+}
+
+func addItem(server: HttpServer) {
+    server.post["/items"] = { request in
+        let params = paramsObject(with: request)
+        guard let title = params["title"] else { return .BadRequest(.Json(["error": "title can't be blank"])) }
+        
+        let newItem = Item(title: title, createdAt: NSDate())
```
Comment (kittinunf) [inline SecretListServer/SecretListServer/Auth.swift]: This line might cause problem as well na.

File: SecretListServer/SecretListServer/Auth.swift [hunk 2]
```diff
@@ -12,17 +12,49 @@ import Swifter
...
-      return _map
+var items = [
+  Item(title: "Buy Milk", createdAt: NSDate()),
+  Item(title: "Sing Karaoke", createdAt: NSDate()),
+  Item(title: "Test drive", createdAt: NSDate()),
+  Item(title: "Visit mom", createdAt: NSDate()),
+  Item(title: "Bake a cake", createdAt: NSDate()),
+]
+
+func allItems(server: HttpServer) {
```
Comment (kittinunf) [inline SecretListServer/SecretListServer/Auth.swift]: These 2 functions are supposed to be in a separate file la.

`Auth.swift`
`Item.swift`

**Comment by macbaszii** [comment]
@kittinunf done, refactoring. Could you review and merge ? Thanks.

**Comment by kittinunf** [comment]
Can you split item related API out into its own file?

**Comment by macbaszii** [comment]
@kittinunf done by https://github.com/macbaszii/iDeveloper-MVVM/pull/10/commits/f35d491130602961f8a0c90a4fa2978ea147d3bb

**Comment by kittinunf** [comment]
Nice! LGTM!

**Pull request merged**